include/mbgl/map/camera.hpp

```diff
@@ -96,4 +98,48 @@ struct AnimationOptions {
         : duration(d) {}
 };
 
+/** Various options for accessing physical properties of the underlying camera entity.
+    A direct access to these properties allows more flexible and precise controlling of the camera
+    while also being fully compatible and interchangeable with CameraOptions. All fields are optional. */
+struct FreeCameraOptions {
+    using vec3 = std::array<double, 3>;
+    using vec4 = std::array<double, 4>;
```

Comment:
these do not have to be nested types, they are quite generic

Reply:
Yes you're right. Originally I wanted to keep the separation between public and private vector types but there's really no need for that as they're just type aliases.